File: app/src/main/java/com/example/pratiler/models/Livro.java
package com.example.pratiler.models;

public class Livro {
    String titulo;
    String sinopse;
    String autor;
    String capa;

    public Livro(String titulo, String sinopse, String autor, String capa){
        this.titulo = titulo;
        this.sinopse = sinopse;
        this.autor = autor;
        this.capa = capa;
    }

    public String getTitulo() {
        return titulo;
    }

    public String getSinopse() {
        return sinopse;
    }

    public String getAutor() {
        return autor;
    }

    public String getCapa() {
        return capa;
    }

    public void setTitulo(String titulo) {
        this.titulo = titulo;
    }

    public void setSinopse(String sinopse) {
        this.sinopse = sinopse;
    }

    public void setAutor(String autor) {
        this.autor = autor;
    }

    public void setCapa(String capa) {
        this.capa = capa;
    }
}

File: hibrido/lib/models/livro.dart
class Livro {
  final int id;
  final String titulo;
  final String autor;
  final String sinopse;
  final int nPaginas;
  final String capa;

  Livro({
    required this.id,
    required this.titulo,
    required this.autor,
    required this.sinopse,
    required this.nPaginas,
    required this.capa,
  });

  static Livro fromJson(Map<String, dynamic> json) {
    return Livro(
      id: json['id'],
      titulo: json['titulo'],
      autor: json['autor']['nome'],
      sinopse: json['sinopse'],
      nPaginas: json['n_paginas'],
      capa: json['capa'],
    );
  }

  Map<String, dynamic> toSql() {
    return { 'id': this.id, 'titulo': this.titulo, 'autor': this.autor, 'sinopse': this.sinopse, 'nPaginas': this.nPaginas, 'capa': this.capa }
  }
}
